### PR TITLE
Use Sentry instead of Fabric

### PR DESCRIPTION
**Pull request by jkmassel**
**To Test:**

1) Check out this branch
2) Run `cd ~/.mobile-secrets && git pull && cd - && bundle exec fastlane run configure_apply` to update your local configuration secrets.
3) Download [the attached patch file](https://github.com/woocommerce/woocommerce-ios/files/3234545/0001-Add-crash-logging-tests.txt), and run `git apply $path_to_file`.
4) In Xcode, do a workspace search for `shouldSendEvent(_` and choose the first result. Comment out the debug-specific code so that it looks like this:
```swift
//        #if DEBUG
//        let result = false
//        #else
        let result = !CrashLogging.userHasOptedOut
//        #endif
```

5) Run the app on-device.
6) Disconnect the device from the host machine, and force-quit the app.
7) Relaunch the app and Log in, ensuring that your user is set up to send crash reports (by default, this will be disabled).
8) Open Settings. Note that a new test event shows up in Sentry almost immediately  – you can see it under "Events" if you filter by woocommerce-ios. It should have your username associated with it.
9) Navigate to Settings > Feature Request (the app will crash)
10) Re-run the app, and let it idle for a moment (this'll ensure the crash report goes to Sentry)
11) Note your shiny new crash event in Sentry (it may take ~60 seconds to show up) – you can see it under "Events" if you filter by woocommerce-ios. It should have your username associated with it, as well as a full stack trace, and the device locale.

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by mindgraffiti** [comment]
@jkmassel when you're ready to request a review again, I'd suggest adding @astralbodies in addition / instead of me. He's working on the global opt-out feature and this PR is going to deeply affect that work.

**Comment by jkmassel** [comment]
Thanks @mindgraffiti – will do!

**Comment by astralbodies** [comment]
Added the `[Status] Not Ready for Review` label until it's ready for review.

**Comment by jkmassel** [comment]
@astralbodies – this is ready for review! 😄

**Comment by mindgraffiti** [comment]
@jkmassel please assign a reviewer next time so we know it's ready.

**Comment by jkmassel** [comment]
Shoot, sorry – I didn't realize that'd been missed!

File: WooCommerce/Classes/Extensions/UserDefaults+Woo.swift [hunk 1]
```diff
@@ -33,6 +33,12 @@ extension UserDefaults {
         return value(forKey: key.rawValue) as? T
     }
 
+    /// Returns the boolean value (if any) associated with the specified Key.
+    ///
+    func bool(forKey key: Key) -> Bool? {
+        return bool(forKey: key.rawValue)
+    }
+
```
Comment (mindgraffiti) [inline WooCommerce/Classes/Extensions/UserDefaults+Woo.swift]: You can delete this. With the power of [generics](https://docs.swift.org/swift-book/LanguageGuide/Generics.html), it's already been defined above on Line 32. Pretty nifty!

Comment (jkmassel) [reply WooCommerce/Classes/Extensions/UserDefaults+Woo.swift]: Ah right! Thanks!!

Fixed in 9a2690f366374cef604a6083f828b689fb299502

**Comment by loremattei** [comment]
I'm moving this to 2.1 because 2.0 has been cut. Feel free to move it back to 2.0 and to target the release branch before merging it.

**Comment by mindgraffiti** [comment]
Nudging @astralbodies who has final say on this PR, since this affects his current work with the global opt-out setting.

**Comment by astralbodies** [comment]
I'm going to say it's good to merge. @ctarda took over that opt-out flag. Let's merge this and then Cesar and refactor my crappy code to account for the changes.

**Pull request merged**